### ISSUE TITLE
Duplicate server button / endpoint

### ISSUE THREAD

**Issue by arantesxyz** [pterodactyl/panel#2148]
**Describe the solution you'd like**
I'd love to have a way to duplicate a server directly on the node i'm with all it's files, etc.

**Describe alternatives you've considered**
I currently use the pterodactyl api to duplicate the server info and then the sftp to do the same with the files. But I'd like a way of duplicating the files directly on the node, no need to make an sftp transfer if the server will be on the same node.

**Comment by DaneEveritt** [comment]
Somewhat duplicative of https://github.com/pterodactyl/panel/issues/2059 although I do see a difference between what you're asking and that.

**Comment by arantesxyz** [comment]
Yes, the idea is the same. But'd like if you guys added a way to also copy the files, and if they're inside the same node, that could be done directly on the file system so, no need to connect using the sftp.

I'd gladly send a pr with those changes, but i unfortunately have no idea how wings (the go version) work at the moment.

**Comment by Sir3lit** [comment]
isn't this what packs are kinda for?

**Comment by arantesxyz** [comment]
> isn't this what packs are kinda for?

Idk, where i can find documentation for packs? I couldn’t find any!

**Comment by DaneEveritt** [comment]
Duplicate of #481 but closing that one in favor of this one since there is a little more linking going on here.

**Comment by ad1tya2** [comment]
Why is this stagnant,,,
this feature would really help out all people and Will make it a lot easier for us to manage and make servers

**Comment by matthewpi** [comment]
> Why is this stagnant,,,
> this feature would really help out all people and Will make it a lot easier for us to manage and make servers

Compared to other features that have been requested, this isn't a high priority addition.  Due to limited development time, we prioritize specific features and don't always have time to implement ones that can take a long-period of time.  Till this day, there is still an open issue asking for admin permissions which was opened in mid-2017 and still hasn't been implemented which is arguably more important than this.  Not to mention, that only a small subset of people will benefit from this addition compared to many of the other feature requests that are currently open.

**Comment by DaneEveritt** [comment]
No one is paying me to develop this feature, so it'll happen when I decide I am excited to build it out. Until such time, I am focusing my limited free time on building things that are requested by those who are sponsoring the project right now.

**Comment by FallenSuperr** [comment]
> No one is paying me to develop this feature, so it'll happen when I decide I am excited to build it out. Until such time, I am focusing my limited free time on building things that are requested by those who are sponsoring the project right now.

Are we allowed to pay for this feature to be developed? If so I'd appreciate if you contacted my Discord at Fallen#1200.

**Comment by DaneEveritt** [comment]
You're welcome to Sponsor the project using the "Sponsor" button at the top of this repository if you'd like to contribute monetarily and have your requests prioritized with other sponsor requests. Other than that there is no specific "pay $X and have Y feature developed at the front of the line" system with this project.

**Comment by untotren** [comment]
+1 on this!  Right now for my host I have to download a backup, build a new server, upload the backup.zip then uncompress on the new server.  So this takes 1-2 hours simply due to bandwidth of downloading and uploading over a residential connection.  Thank you for what you've done already with this panel, incredible, you're an artistic genius.

**Comment by ad1tya2** [comment]
I think some of the transfer server code can also be used in this too,
This would be a really good addition that would help a lot of people.

**Comment by kubo6472** [comment]
+1 also, can we currently restore backups to different servers? It does not save as much time as directly duplicating it, but saves some definitely.

**Comment by christiansenandi** [comment]
+1, i would love this feature as well!

**Comment by TheDarthFather** [comment]
+1 
Use case: copy a production server to a test one, so you can test an upgrade/dev changes.

**Comment by Fumio-26** [comment]
bump

**Comment by Fumio-26** [comment]
+1

**Comment by fucksophie** [comment]
way easier method if you're self-hosting pterodactyl, you can just go into /var/lib/pterodactyl/volumes and copy over the volume

**Comment by HerrSammyDE** [comment]
When will this cool function arrive?

Automatic transfers would be great :D

**Comment by NicolasGoeddel** [comment]
Alternatively it would be cool to be able to create a server from a backup of another one.

**Comment by dankden** [comment]
> +1 Use case: copy a production server to a test one, so you can test an upgrade/dev changes.

this would also give me a bit of confidence in using the transfer between nodes function. Would allow to minimize downtime if you can keep one up and then once one transfers can fully migrate.

**Comment by ReconOfficial** [comment]
> Alternatively it would be cool to be able to create a server from a backup of another one.

this, i don't know why it's been 8 years (since 2017) of this feature being requested and it has yet to be added.

**Comment by FreshImmuc** [comment]
+1

**Comment by DaneEveritt** [comment]
In the interest of maintainer sanity and cleaning up an enormous backlog of stale feature requests we have moved to requiring all feature requests be discussed within [Discussions](https://github.com/orgs/pterodactyl/discussions/categories/feature-requests). Please consider re-opening this request there if you believe it is still relevant and worth consideration.

See https://github.com/pterodactyl/panel/issues/5491 for additional context.

Thanks.

<sub>_This comment is in the interest of issue tracker maintenance: it does not imply that maintainers are endorsing or dismissing your request._</sub>
